src/main/java/btdex/core/BurstNode.java

```diff
@@ -210,35 +210,39 @@ public void run() {
 						AssetOrder[] asks = NS.getAskOrders(m.getTokenID()).blockingGet();
 						AssetOrder[] bids = NS.getBidOrders(m.getTokenID()).blockingGet();
 						
-						BurstValue lowestPrice = null;
-						BurstValue highestPrice = null;
-						Date past24h = new Date(System.currentTimeMillis() - 24 * 3600_000);
-						BurstValue baseVolume = BurstValue.fromPlanck(0L);
-						BurstValue quoteVolume = BurstValue.fromPlanck(0L);
-						double priceChangePerc = 0.0;
-						for(AssetTrade t : trades) {
-							if(t.getTimestamp().getAsDate().before(past24h)) {
-								priceChangePerc = 100d*(trades[0].getPrice().doubleValue() - t.getPrice().doubleValue())/trades[0].getPrice().doubleValue();
-								break;
+						// Handle delayed trades data
+						if(trades!=null && trades.length>0) {
+							BurstValue lowestPrice = null;
+							BurstValue highestPrice = null;
+							Date past24h = new Date(System.currentTimeMillis() - 24 * 3600_000);
+							BurstValue baseVolume = BurstValue.fromPlanck(0L);
+							BurstValue quoteVolume = BurstValue.fromPlanck(0L);
+							double priceChangePerc = 0.0;
+							for(AssetTrade t : trades) {
+								if(t.getTimestamp().getAsDate().before(past24h)) {
+									priceChangePerc = 100d*(trades[0].getPrice().doubleValue() - t.getPrice().doubleValue())/trades[0].getPrice().doubleValue();
+									break;
+								}
+								if(lowestPrice == null || t.getPrice().compareTo(lowestPrice) < 0)
+									lowestPrice = t.getPrice();
+								if(highestPrice == null || t.getPrice().compareTo(highestPrice) > 0)
+									highestPrice = t.getPrice();
+								
+								baseVolume = baseVolume.add(t.getQuantity());
+								quoteVolume = quoteVolume.add(t.getQuantity().multiply(t.getPrice().longValue()));
 							}
-							if(lowestPrice == null || t.getPrice().compareTo(lowestPrice) < 0)
-								lowestPrice = t.getPrice();
-							if(highestPrice == null || t.getPrice().compareTo(highestPrice) > 0)
-								highestPrice = t.getPrice();
-							
-							baseVolume = baseVolume.add(t.getQuantity());
-							quoteVolume = quoteVolume.add(t.getQuantity().multiply(t.getPrice().longValue()));
+								lowestPrice24h.put(m, lowestPrice != null ? lowestPrice : trades[0].getPrice());
+								highestPrice24h.put(m, highestPrice != null ? highestPrice : trades[0].getPrice());
+							baseVolume24h.put(m, baseVolume);
+							quoteVolume24h.put(m, quoteVolume);
+							priceChangePerc24h.put(m, priceChangePerc);
+
+							assetBalances.put(m, balance);
+							assetTrades.put(m, trades);
+							askOrders.put(m, asks);
```

Comment:
variables, asks and bids, are similar. So, we might do something similar with them. The only difference is that we haven't gotten any exceptions with them. That's why I omitted to change the code.

Reply:
Yes, this code is relatively new, part of the API for trade history required by coinmarketcap.